hpat/hiframes_typed.py

```diff
@@ -751,6 +754,96 @@ def f(arr, on_arr, w, center):  # pragma: no cover
             return nodes
         return [assign]
 
+    def _handle_series_combine(self, assign, lhs, rhs, series_var):
+        """translate s1.combine(s2,lambda x1,x2 :...) to prange()
+        """
+        func = guard(get_definition, self.func_ir, rhs.args[1])
```

Comment:
The number of arguments and the type of the first argument should be checked here as well.